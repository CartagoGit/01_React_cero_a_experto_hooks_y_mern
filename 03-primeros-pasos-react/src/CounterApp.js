import React, {useState} from "react";
import PropTypes from "prop-types";

const CounterApp = ({value = 0}) => {

    // const state = useState('Goku');
    // const [nombre, setNombre] = useState('Goku');
    // console.log(nombre, setNombre);

    const [counter, setCounter] = useState(value);


    const handleAdd = ()=>{
        // setCounter(counter+1);
        setCounter( (c) => c+1 );
    }
    const handleReset = () => setCounter(value);
    const handleSub = () => setCounter( counter -1);
    

	return (
		<>
			<h1>CounterApp</h1>
			<h2> {value} </h2>
            <h2> {counter} </h2>

            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSub}>-1</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number
};

export default CounterApp;
