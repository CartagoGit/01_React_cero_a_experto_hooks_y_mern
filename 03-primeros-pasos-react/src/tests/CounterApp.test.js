import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import CounterApp from "../CounterApp";

describe("Pruebas con CounterAPP", () => {
    let wrap;
    beforeEach(()=>{
        wrap = shallow(<CounterApp />);
        // console.log(wrap.text());
    });
	test("Deberia mostrar counterapp correctamente con un snapshoot", () => {
		// const valor = 123;
		// const wrapper = shallow(<CounterApp />);

		expect(wrap).toMatchSnapshot();
	});

	test("Deberia mostrar el valor por defecto 100", () => {
		const valor = 100;
		const wrap = shallow(<CounterApp value={valor} />);
		wrap.find("h2").forEach((node) => {
            // console.log(node.text());
			expect(parseInt(node.text().trim())).toBe(valor);
		});
	});

    
    test('Debe incrementar el contador con el boton +1', () => {
        wrap.find('button').at(0).simulate('click');
        expect(parseInt(wrap.find("h2").at(1).text().trim())).toBe(1);

    });

    test('Debe decrementar el contador con el boton -1', () => {
        wrap.find('button').at(2).simulate('click');
        expect(parseInt(wrap.find("h2").at(1).text().trim())).toBe(-1);

    });

    test('Debe comprobar que el boton reset coloca el valor por defecto', () => {
        const valor = 100;
        const wrap = shallow(<CounterApp value={valor} />);
        wrap.find('button').at(0).simulate('click');
        
        wrap.find('button').at(0).simulate('click');
        wrap.find('button').at(0).simulate('click');
        wrap.find('button').at(1).simulate('click');

        const counterNum = parseInt(wrap.find("h2").at(1).text().trim());

        expect (counterNum).toBe(valor)

    })
    
    
});
