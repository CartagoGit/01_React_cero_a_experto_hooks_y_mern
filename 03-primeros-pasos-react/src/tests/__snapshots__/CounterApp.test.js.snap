// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con CounterAPP Deberia mostrar counterapp correctamente con un snapshoot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    0
     
  </h2>
  <h2>
     
    0
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
